github-comment-fetcher/src/main/java/com/tzachz/commentcounter/apifacade/jsonobjects/GHComment.java

```diff
@@ -15,42 +16,75 @@
 @JsonIgnoreProperties(ignoreUnknown=true)
 public class GHComment {
 
-    private final GHUser user;
-    private final String pullRequestUrl;
-    private final String body;
-    private final String htmlUrl;
-    private final Date createDate;
-
-    @JsonCreator
-    public GHComment(@JsonProperty("user") GHUser user,
-                     @JsonProperty("pull_request_url") String pullRequestUrl,
-                     @JsonProperty("body") String body,
-                     @JsonProperty("html_url") String htmlUrl,
-                     @JsonProperty("created_at") Date createDate) {
-        this.body = body;
-        this.user = user;
-        this.pullRequestUrl = pullRequestUrl;
-        this.htmlUrl = htmlUrl;
-        this.createDate = createDate;
-    }
+    private GHUser user;
```

Comment:
I switched to the setter style because issue comments and review comments from GitHub's API have some overlapping fields but issue comments don't have `pull_request_url` (see the `setIssueUrl` method I added, which is kind of hacky).

I can try using 2 different classes that implement the same interface at the possible cost of duplicate code for the fields.

> I kinda prefer the immutable version. Although it creates a longer argument list for the constructor (downside), it prevents unintended changes to the object after it's created, which might introduce a whole new level of temporal-coupling-related bugs. Any specific reason to make this change?